core/vm/evm.go

```diff
@@ -410,6 +411,7 @@ type codeAndHash struct {
 
 func (c *codeAndHash) Hash() common.Hash {
 	if c.hash == (common.Hash{}) {
+		// when calculating CREATE2 address, we use Keccak256 not Poseidon
```

Comment:
`when calculating CREATE/CREATE2 address`

Reply:
This code hash is only used for `CREATE2`. `CREATE` uses the caller address and nonce.